genie-web/src/main/java/com/netflix/genie/web/spring/actuators/RejoinLeaderElectionActuator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.spring.actuators;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.genie.web.services.ClusterLeaderService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.WriteOperation;
+
+import java.util.Map;
+
+
+/**
+ * An actuator endpoint that exposes leadership status.
+ * Allows leaving and rejoining the election. Useful when a specific set of nodes should be given priority to win the
+ * leader election (e.g., because they are running newer code).
+ *
+ * @since 4.0.0
+ * @author mprimi
+ */
+@Endpoint(id = "rejoinLeaderElection")
+@Slf4j
+public class RejoinLeaderElectionActuator {
```

Comment:
Why not just `LeaderActuator`? this makes it seem like a verb more than a noun

Reply:
This way of creating an actuator using `@WriteOperation` is very nice, but the downside is that there's only 1 possible action associated to a `POST` request. In this case it's 'rejoin'.

If we wanted to add other actions like leave permanently, it would have to be a similar but different controller. Hence the name.

Let me see what i can do in this regards..